coins/bitcoin/src/rpc.rs

```diff
@@ -0,0 +1,538 @@
+        let args = handle_defaults(&mut ext_args, &[null()]);
+        let tx = self
+            .rpc_call::<GetTransactionResult>("gettransaction", &args)
+            .await?;
+        Ok(tx)
+    }
+
+    pub async fn get_transactions(
+        &self,
+        tx_hashes: Vec<&str>,
+    ) -> anyhow::Result<Vec<GetTransactionResult>> {
+        let mut transactions = Vec::<GetTransactionResult>::new();
+        for one_tx in tx_hashes.iter() {
+            let mut ext_args = [into_json(one_tx)?];
+            let args = handle_defaults(&mut ext_args, &[null()]);
+            let one_transaction = self
+                .rpc_call::<GetTransactionResult>("gettransaction", &args)
+                .await?;
+            transactions.push(one_transaction);
+        }
+        Ok(transactions)
+    }
+
+    pub async fn get_raw_transactions(
+        &self,
+        tx_hashes: Vec<&str>,
+        block_hash : Option<&BlockHash>,
+    ) -> anyhow::Result<Vec<Transaction>> {
+        let mut transactions = Vec::<Transaction>::new();
+        for one_tx in tx_hashes.iter() {
+            let one_transaction = self.get_raw_transaction(&bitcoin::Txid::from_str(one_tx).unwrap(), None, block_hash).await.unwrap();
+            transactions.push(one_transaction);
+        }
+        Ok(transactions)
+    }
+
+    pub async fn is_confirmed(&self, tx_hash: &str) -> anyhow::Result<bool> {
+        let tx_block_number = self.get_transaction_block_number(&tx_hash).await?;
+        Ok((self.get_height().await?.saturating_sub(tx_block_number) + 1) >= 10)
+    }
+
+    pub async fn get_transaction_block_number(&self, tx_hash: &str) -> anyhow::Result<usize> {
+        let mut ext_args = [into_json(tx_hash)?];
+        let args = handle_defaults(&mut ext_args, &[null()]);
+        let tx = self
+            .rpc_call::<GetTransactionResult>("gettransaction", &args)
+            .await?;
+        Ok(tx.info.blockheight.unwrap() as usize)
+    }
+
+    pub async fn get_block_hash(&self, height: usize) -> Result<bitcoin::BlockHash> {
+        let mut ext_args = [into_json(height)?];
+        let args = handle_defaults(&mut ext_args, &[null()]);
+        let info: bitcoin::BlockHash = self
+            .rpc_call::<bitcoin::BlockHash>("getblockhash", &args)
+            .await?;
+        Ok(info)
+    }
+
+    pub async fn get_block_transactions(
+        &self,
+        height: usize,
+    ) -> anyhow::Result<Vec<GetTransactionResult>> {
+        let block_hash = self.get_block_hash(height).await.unwrap();
+        let block_info = self.get_block(&block_hash).await.unwrap();
+        let tx_ids: Vec<String> = block_info.txdata.iter().map(|one_tx| one_tx.txid().to_string()).collect();
+        
+        let tx_ids_str: Vec<&str> = tx_ids.iter().map(|s| &s[..]).collect();
+        let transactions = self.get_transactions(tx_ids_str).await.unwrap();
+        Ok(transactions)
+    }
+
+    pub async fn get_block_raw_transactions(
```

Comment:
Why do we have get_raw_transaction, get_block_raw_transactions, get_block, get_block_with_transactions, AND get_raw_transactions? Can we *please* just consolidate these.

The processor expects/supports either:
- Block with TXs
- Block without TXs

The former is superior and overall more efficient. The only reason we support the latter is for a Monero consideration/API boundary. Can we just have `get_block`, returning a bitcoin type with all its TXs, please? If we technically need to define a block as a block hash, and from there have get_block_transactions, fine, I just want to minimize this file.